metricproviders/newrelic/newrelic.go

```diff
@@ -155,15 +155,20 @@ func NewNewRelicAPIClient(metric v1alpha1.Metric, kubeclientset kubernetes.Inter
 
 	newrelicOptions := []newrelic.ConfigOption{newrelic.ConfigPersonalAPIKey(apiKey), newrelic.ConfigUserAgent(userAgent)}
 
-	// support either base-url or region; default to us region if neither is used
-	if _, ok := secret.Data["base-url"]; ok {
-		newrelicOptions = append(newrelicOptions, newrelic.ConfigBaseURL(string(secret.Data["base-url"])))
-	} else {
-		region := "us"
-		if _, ok := secret.Data["region"]; ok {
-			region = string(secret.Data["region"])
-		}
-		newrelicOptions = append(newrelicOptions, newrelic.ConfigRegion(region))
+	region := "us"
+	if _, ok := secret.Data["region"]; ok {
+		region = string(secret.Data["region"])
+	}
+	newrelicOptions = append(newrelicOptions, newrelic.ConfigRegion(region))
+
+	// base URL for the new relic REST API
+	if _, ok := secret.Data["base-url-rest"]; ok {
+		newrelicOptions = append(newrelicOptions, newrelic.ConfigBaseURL(string(secret.Data["base-url-rest"])))
+	}
+
+	// base URL for the nerdgraph (graphQL) API
+	if _, ok := secret.Data["base-url-nerdgraph"]; ok {
+		newrelicOptions = append(newrelicOptions, newrelic.ConfigNerdGraphBaseURL(string(secret.Data["base-url-nerdgraph"])))
```

Comment:
this is a change from the original approach, based on reading the docs. Looks like we need to define separate REST API and NerdGraph (aka GraphQL) API base URLs in addition to the region.

Reply:
there are some other base URL config keys available but these are the only two that should be relevant for this specific application